lib/Backend/BackendUtils.cpp

```diff
@@ -187,9 +187,29 @@ runtime::RuntimeBundle::create(const IRFunction &F,
   }
   auto constantMaxSize = constantAllocator.getMaxMemoryUsage();
 
-  // Compute the offsets for Placeholders.
+  // Placeholders should be allocated in a order of Input|InputOutput|Output.
+  std::vector<Placeholder const *> inputOutputPlaceholders;
+  std::vector<Placeholder const *> outputPlaceholders;
+  std::vector<Placeholder const *> contiguousPlaceholders;
   for (auto &v : F.findPlaceholders()) {
-    // Get the WeightVar for each Placeholder to calculate offsets.
+    if (isInput(v)) {
+      if (!isOutput(v)) {
+        contiguousPlaceholders.push_back(v);
+      } else {
+        inputOutputPlaceholders.push_back(v);
+      }
+    } else {
+      outputPlaceholders.push_back(v);
+    }
+  }
+  for (auto &v : inputOutputPlaceholders) {
```

Comment:
Rather than creating multiple vectors, it may be cleaner to move the body of the loop below (that does the allocation & insertion into the symbol table) into a helper function and then loop through your inputs, inputsOutputs and outputs containers calling that helper.

Reply:
> Rather than creating multiple vectors, it may be cleaner to move the body of the loop below (that does the allocation & insertion into the symbol table) into a helper function and then loop through your inputs, inputsOutputs and outputs containers calling that helper.

OK. Moved them all into a helper function.